src/test/scala/io/citrine/lolo/bags/BaggedResultTest.scala

```diff
@@ -31,44 +31,87 @@ class BaggedResultTest {
+    */
   @Test
   def testBaggedSingleResultGetUncertainty(): Unit = {
     val noiseLevel = 100.0
     val rng = new Random(237485L)
-    Seq(RegressionTreeLearner(),GuessTheMeanLearner()).foreach{ baseLearner =>
-      Seq(30,100,301).foreach { nRows =>
-        val trainingDataTmp = TestUtils.generateTrainingData(nRows, 1, noise = 0.0, function = _ => 0.0, seed = rng.nextLong())
-        val trainingData = (trainingDataTmp).map { x => (x._1, x._2 + noiseLevel * rng.nextDouble()) }
-        val baggedLearner = Bagger(baseLearner, numBags = 2 * nRows, uncertaintyCalibration = true)
-        val RFMeta = baggedLearner.train(trainingData)
-        val RF = RFMeta.getModel()
-        val results = RF.transform(trainingData.take(1).map(_._1))
-
-        val sigmaObs: Seq[Double] = results.getUncertainty().get.asInstanceOf[Seq[Double]]
-//        sigmaMean.zip(results.asInstanceOf[RegressionResult].getStdDevMean().get).foreach{ case (a,b) =>
-//          assert(a == b, "Expected getUncertainty(observational=false)=getStdDevMean()")
-//        }
-//        val sigmaMean: Seq[Double] = results.getUncertainty(observational = false).get.asInstanceOf[Seq[Double]]
-        sigmaObs.zip(results.asInstanceOf[RegressionResult].getStdDevObs().get).foreach{ case (a,b) =>
-          assert(a == b, "Expected getUncertainty()=getStdDevObs()")
-        }
 
-//        sigmaObs.zip(sigmaMean).foreach { case (sObs, sMean) => assert(sObs > sMean, "Uncertainty should be greater when observational = true.") }
-
-        // We have strong theoretical guarantees on the behavior of GuessTheMeanLearner, so let's exercise them.
-        // TODO(grobinson): implement similar test for RegressionTreeLearner.
-        if (baseLearner.isInstanceOf[GuessTheMeanLearner]) {
-          // NOTE: these bounds reflect a ~3x systematic variance under-estimation in this particular test setting.
-          val rtolLower = 5.0  // Future recalibration should decrease this number.
-          val rtolUpper = 1.0  // Future recalibration should increase this number.
-          sigmaObs.foreach { s =>
-            assert(rtolLower * s > noiseLevel, "Observational StdDev getUncertainty() is too small.")
-            assert(s < rtolUpper * noiseLevel, "Observational StdDev getUncertainty() is too large.")
+    Seq(RegressionTreeLearner(), GuessTheMeanLearner()).foreach { baseLearner =>
+      // These are in Seqs as a convenience for repurposing this test as a diagnostic tool.
+      Seq(128).foreach { nRows =>
+        Seq(16).foreach { nCols =>
+          Seq(2).map { n => n * nRows }.foreach { nBags =>
+            // Used for error output.
+            val configDescription = s"learner=${baseLearner.getClass().toString()}\tnRows=$nRows\tnCols=$nCols\tnumBags=$nBags"
+
+            // Count times getStdDevObs is > and < getStdDevMean
+            var countStdObsGtStdMean = 0.0
+            var countStdObsLtStdMean = 0.0
+
+            (1 to 10).foreach { _ =>
```

Comment:
Not blocking, but in order to avoid `var` you could do
```
val sigmaObsAndMean: Seq[Seq[(Double, Double)]] = (1 to 10).map {_ =>
```
then use `flatMap` to turn it into a `Seq[(Double, Double)]` and use `.count` to count the number for which `sObs > sMean`. You could also do the bounds checks on this sequence.

Reply:
Cool, done. (Mostly to improve my Scala idioms.)